packages/replay-next/components/sources/SyntaxHighlitedLine.test.tsx

```diff
@@ -0,0 +1,43 @@
+import renderer from 'react-test-renderer';
+import SyntaxHighlightedLine from "./SyntaxHighlightedLine";
+
+const SNIPPET = 'button.addEventListener("click", () => onClick(button, index + 1));';
+const SNIPPET_TOKENS = [
+  { columnIndex: 4, types: ["variableName"], value: "button" },
+  { columnIndex: 10, types: ["operator"], value: "." },
+  { columnIndex: 11, types: ["propertyName"], value: "addEventListener" },
+  { columnIndex: 27, types: ["punctuation"], value: "(" },
+  { columnIndex: 28, types: ["string"], value: '"click"' },
+  { columnIndex: 35, types: ["punctuation"], value: "," },
+  { columnIndex: 36, types: null, value: " " },
+  { columnIndex: 37, types: ["punctuation"], value: "(" },
+  { columnIndex: 38, types: ["punctuation"], value: ")" },
+  { columnIndex: 39, types: null, value: " " },
+  { columnIndex: 40, types: ["punctuation"], value: "=>" },
+  { columnIndex: 42, types: null, value: " " },
+  { columnIndex: 43, types: ["variableName"], value: "onClick" },
+  { columnIndex: 50, types: ["punctuation"], value: "(" },
+  { columnIndex: 51, types: ["variableName"], value: "button" },
+  { columnIndex: 57, types: ["punctuation"], value: "," },
+  { columnIndex: 58, types: null, value: " " },
+  { columnIndex: 59, types: ["variableName"], value: "index" },
+  { columnIndex: 64, types: null, value: " " },
+  { columnIndex: 65, types: ["operator"], value: "+" },
+  { columnIndex: 66, types: null, value: " " },
+  { columnIndex: 67, types: ["number"], value: "1" },
+  { columnIndex: 68, types: ["punctuation"], value: ")" },
+  { columnIndex: 69, types: ["punctuation"], value: ")" },
+  { columnIndex: 70, types: ["punctuation"], value: ";" },
+];
+
+describe("syntax-highlighting", () => {
+  it("syntax highliting should work without passing tokens to SyntaxHighlitedLine", async () => {
+    const tree = renderer.create(<SyntaxHighlightedLine code={SNIPPET} />).toJSON()
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("syntax highliting should work when tokens are passed to SyntaxHighlitedLine", async () => {
+    const tree = renderer.create(<SyntaxHighlightedLine code={SNIPPET} tokens={SNIPPET_TOKENS} />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+});
```

Comment:
I understand why you're using react-test-library for this, but we have an existing convention for e2e testing in the replay-next package and I don't think we should introduce a _new_ way of writing these tests just for a single case (particularly without discussion).

We appreciate new tests! If you'd like to add one, check out the README in this directory for more information about how we write this type of test:
https://github.com/replayio/devtools/tree/main/packages/replay-next/playwright

I also think it's okay in this case to rely on existing test coverage to prevent a regression.

Reply:
Thanks @bvaughn - this is on me.
I recommended some random resources on snapshot testing earlier and pointed to some of your existing tests, but evidently I pointed in the wrong direction 👀

Reply:
@Domiii Unit vs e2e tests depends on the type of change.

Reply:
> I understand why you're using react-test-library for this, but we have an existing convention for e2e testing in the replay-next package and I don't think we should introduce a _new_ way of writing these tests just for a single case (particularly without discussion).
> 
> We appreciate new tests! If you'd like to add one, check out the README in this directory for more information about how we write this type of test: https://github.com/replayio/devtools/tree/main/packages/replay-next/playwright
> 
> I also think it's okay in this case to rely on existing test coverage to prevent a regression.

@bvaughn got it, so I'll just remove the tests I have added for now.